RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -133,1112 +137,446 @@ void AntiElectronIDMVA6::beginEvent(const edm::Event& evt, const edm::EventSetup
-                                    Float_t ElecMvaInDeltaEta) {
+template <class TauType, class ElectronType>
+double AntiElectronIDMVA6<TauType, ElectronType>::MVAValue(const TauVars& tauVars,
+                                                           const TauGammaMoms& tauGammaMoms,
+                                                           const ElecVars& elecVars) {
   if (!isInitialized_) {
-    throw cms::Exception("ClassNotInitialized") << " AntiElectronMVA not properly initialized !!\n";
+    throw cms::Exception("ClassNotInitialized") << " AntiElectronMVA6 not properly initialized !!\n";
   }
 
   double mvaValue = -99.;
 
   const float ECALBarrelEndcapEtaBorder = 1.479;
-  float ElecDeltaPinPoutOverPin = (ElecPin > 0.0) ? (std::abs(ElecPin - ElecPout) / ElecPin) : 1.0;
-  float ElecEecalOverPout = (ElecPout > 0.0) ? (ElecEecal / ElecPout) : 20.0;
-  float ElecNumHitsDiffOverSum = ((ElecGSFNumHits + ElecKFNumHits) > 0.0)
-                                     ? ((ElecGSFNumHits - ElecKFNumHits) / (ElecGSFNumHits + ElecKFNumHits))
-                                     : 1.0;
-
-  if (deltaR(TauEtaAtEcalEntrance, TauPhi, ElecEta, ElecPhi) > 0.3 && TauSignalPFGammaCandsIn == 0 && TauHasGsf < 0.5) {
-    if (std::abs(TauEtaAtEcalEntrance) < ECALBarrelEndcapEtaBorder) {
-      Var_NoEleMatch_woGwoGSF_Barrel_[0] = TauEtaAtEcalEntrance;
-      Var_NoEleMatch_woGwoGSF_Barrel_[1] = TauLeadChargedPFCandEtaAtEcalEntrance;
-      Var_NoEleMatch_woGwoGSF_Barrel_[2] = std::min(float(2.), TauLeadChargedPFCandPt / std::max(float(1.), TauPt));
-      Var_NoEleMatch_woGwoGSF_Barrel_[3] = std::log(std::max(float(1.), TauPt));
-      Var_NoEleMatch_woGwoGSF_Barrel_[4] = TauEmFraction;
-      Var_NoEleMatch_woGwoGSF_Barrel_[5] = TauLeadPFChargedHadrHoP;
-      Var_NoEleMatch_woGwoGSF_Barrel_[6] = TauLeadPFChargedHadrEoP;
-      Var_NoEleMatch_woGwoGSF_Barrel_[7] = TauVisMassIn;
-      Var_NoEleMatch_woGwoGSF_Barrel_[8] = TaudCrackEta;
-      Var_NoEleMatch_woGwoGSF_Barrel_[9] = TaudCrackPhi;
+  float elecDeltaPinPoutOverPin = (elecVars.pIn > 0.0) ? (std::abs(elecVars.pIn - elecVars.pOut) / elecVars.pIn) : 1.0;
+  float elecEecalOverPout = (elecVars.pOut > 0.0) ? (elecVars.eEcal / elecVars.pOut) : 20.0;
+  float elecNumHitsDiffOverSum =
+      ((elecVars.gsfNumHits + elecVars.kfNumHits) > 0.0)
+          ? ((elecVars.gsfNumHits - elecVars.kfNumHits) / (elecVars.gsfNumHits + elecVars.kfNumHits))
+          : 1.0;
+
+  if (deltaR(tauVars.etaAtEcalEntrance, tauVars.phi, elecVars.eta, elecVars.phi) > 0.3 &&
+      tauGammaMoms.signalPFGammaCandsIn == 0 && tauVars.hasGsf < 0.5) {
+    if (std::abs(tauVars.etaAtEcalEntrance) < ECALBarrelEndcapEtaBorder) {
+      Var_NoEleMatch_woGwoGSF_Barrel_[0] = tauVars.etaAtEcalEntrance;
+      Var_NoEleMatch_woGwoGSF_Barrel_[1] = tauVars.leadChargedPFCandEtaAtEcalEntrance;
+      Var_NoEleMatch_woGwoGSF_Barrel_[2] =
+          std::min(float(2.), tauVars.leadChargedPFCandPt / std::max(float(1.), tauVars.pt));
```

Comment:
```2.f``` instead of ```float(2.)``` wouldn't require type conversion
(the same in several other places here and below)